sections/Chapter 3 Sprint 1/ER_Model.tex

```diff
@@ -0,0 +1,67 @@
+The entity sets are represented graphically with a rectangle and relationship sets connecting the entity sets with a diamond \cite{DBSBook}.
+The attributes associated with the entity- and relationship sets can be modelled using ovals \cite{KatjaFirstPP}, however other alternatives exists.\footnote{Our style of choice is presented in Chapter 6.10 of \citetitle{DBSBook} \textit{\citefield[]{DBSBook}[]{edition}}}
+Similar to the relational model, the entities (tuples) must be uniquely identified by one or more attributes. This is denoted by underlining the attributes. 
+
+The E-R model have notations that denotes the participations each entity have in a connecting relationship \cite{DBSBook}.
+In this project, we will use \text{min-max} notation; this notation denotes the minimum and maximum amount of entities participating in a relationship. 
+
+\begin{figure}[htp]
+    \centering
+    \includegraphics[scale=0.5]{Images/book_example_w_cardinality.png}
+    \caption{E-R diagram of the book and book owner example.}
+    \label{fig:ER_Book_Example}
+\end{figure}
+
+Figure \ref{fig:ER_Book_Example} shows an E-R diagram equivalent to the $book$, $owns$, and $book_owner$ relations from equation \ref{eq:bookOwnerExample} and \ref{eq:relational_schema}.
+In figure \ref{fig:ER_Book_Example} we also see the  the participation cardinalities of the two entity sets. 
+The relationship from $book\_owner$ to $owns$ is one of total participation. That is, each book owner entity must own at least one book.
+\begin{figure}[h]
+    \centering
+    \includegraphics[scale=0.5]{Images/cardinalities.png}
+    \caption{Participation ratios for ER relationships}
+    \label{fig:ERDiagram_Cardinality}
+\end{figure}
+The participation from $book$ to $owns$ is partial. Thus the model represents that books can be in the database, even if no one owns it, that a book can be owned by multiple book owners.
+Participation connections can be seen on figure \ref{fig:ERDiagram_Cardinality}.
+
+
+Other forms of entity sets exits. For instance, a weak entity set, denoted by a an oval with doubled edges, is an entity sets whose existence is based on another entity set. Relationship sets connecting a weak entity set to its' identifying entity set will not have any attributes.
+A weak entity set is identified by its \textit{identifying entity set}'s primary key along with extra attributes. 
+The relationship between weak entity sets and its identifying set is always many-to-one with total  participation of the weak entity set.
+
+Since the relational model is described using relations, we can also describe operations that can be performed on the relations.
+Therefore, it is useful to be able to convert an E-R model into an equivalent relational model, where domains of attributes can be well defined and operations on the entity sets described mathematically.
+
+\subsubsection*{Converting E-R Models to Relations}
```

Comment:
```suggestion
\subsubsection*{Converting ER Models to Relations}
When converting an ER diagram into an equivalent relational model, we must consider the domains of the attributes in the ER model, since these need to be described in the relational model. Having the domain of the relations described makes it possible to establish well-defined operations that can be performed on the relations.
```